tdh/resource_data_plane.go

```diff
@@ -94,55 +96,68 @@ func (r *dataPlaneResource) Schema(ctx context.Context, _ resource.SchemaRequest
-				Required:            true,
+				MarkdownDescription: "Name of Kubernetes Cluster. Please use datasource `tdh_k8s_clusters` to get the list of available clusters from an account.\n Note: This field is non-mandatory during the TAS data plane creation . \n It is a mandatory field during Non TAS (i.e `tkgm`, `tkgs`, `openshift`) data plane creation.",
+				Required:            false,
+				Optional:            true,
 			},
 			"shared": schema.BoolAttribute{
-				Description: "Shared Data Plane",
-				Required:    true,
+				MarkdownDescription: "Shared Data Plane.\n Note: This field should be set to true during the TAS data plane creation . \n It can be set to true/false during Non TAS (i.e `tkgm`, `tkgs`, `openshift`) data plane creation.",
+				Required:            true,
 			},
 			"configure_core_dns": schema.BoolAttribute{
 				Description: "Turn on publishing the TDH's DNS to core DNS of the K8S cluster. This will enable communication between pods of the control plane and the data plane with control plane pods. Disable this only when the TDH base URL has a forwarder set in corporate DNS or for some specific use case.\nPlease note that TDH needs these communications between the pods to function.",
 				Optional:    true,
 			},
 			"auto_upgrade": schema.BoolAttribute{
-				Description: "Whether to enable auto-upgrade on Data plane",
-				Required:    true,
+				MarkdownDescription: "Whether to enable auto-upgrade on Data plane.\n Note: This field should be set to false for TAS data-plane creation ",
+				Optional:            true,
+				Required:            false,
 			},
 			"cp_bootstrapped_cluster": schema.BoolAttribute{
-				Description: "Whether to onboard Data Plane on a K8s cluster running TDH Control Plane",
-				Optional:    true,
+				MarkdownDescription: "Whether to onboard Data Plane on a K8s cluster running TDH Control Plane \n Note: Not a required field during TAS data-plane creation",
+				Optional:            true,
 			},
 			"org_id": schema.StringAttribute{
-				Description: "Organization ID",
+				Description: "Organization ID. This filed is not required during TAS data-plane creation",
 				Optional:    true,
 			},
 			"provider_name": schema.StringAttribute{
 				Description: "Provider name",
-				Computed:    true,
+				Required:    true,
 			},
 			"backup_storage_class": schema.StringAttribute{
-				Description: "Backup Storage Class will be used to create all backups on this data plane. Please note this cannot be changed in future.",
-				Required:    true,
+				MarkdownDescription: "Backup Storage Class will be used to create all backups on this data plane. Please note this cannot be changed in future.\n Note: This field is non-mandatory during the TAS data plane creation. \n It is a mandatory field during  Non TAS (i.e `tkgm`, `tkgs`, `openshift`) data plane creation.",
```

Comment:
there also formatting needs to be proper. To check how it looks open corresponding generated .md file in intelli dual view
![image](https://github.com/svc-bot-mds/terraform-provider-tdh/assets/129835279/05bfecdf-2e77-4859-8f33-0bed43541b65)